src/contract.rs

```diff
@@ -185,9 +208,23 @@ pub fn execute(
             protocol_fee_percent,
             protocol_fee_recipient,
         ),
+        ExecuteMsg::FreezeDeposits {} => execute_freeze_deposits(deps, info.sender),
     }
 }
 
+fn execute_freeze_deposits(deps: DepsMut, sender: Addr) -> Result<Response, ContractError> {
```

Comment:
i think there should be an unfreeze option? Maybe pass a pool with this message?

Reply:
I added boolean in the message.